markdown/src/org/intellij/plugins/markdown/structureView/MarkdownStructureElement.java

```diff
@@ -117,28 +103,63 @@ private ItemPresentation getPresentationImpl() {
   @NotNull
   @Override
   public TreeElement[] getChildren() {
-    final PsiElement parentToTraverse = myElement instanceof MarkdownFile ? myElement.getFirstChild() : myElement;
+    final List<MarkdownStructureElement> childrenElements = new ArrayList<MarkdownStructureElement>();
+
+    PsiElement nextSibling = myElement instanceof MarkdownFile ?
+            myElement.getFirstChild().getFirstChild() :
+            myElement.getNextSibling();
+
+    PsiElement maxContentLevel = null;
+
+    while (nextSibling != null) {
+      if (nextSibling != myElement && isSameLevelOrHigher(nextSibling, myElement)) {
```

Comment:
Can the check `nextSibling != myElement` fail at all? I see that nextSibling is initialized as a child or sibling to `myElement` and only moves further..